### PR TITLE
Improve UnitTest docs

### PR DESCRIPTION
**Pull request by ben-albrecht**
Make several improvements to UnitTest docs:

- no-doc's the error submodule (implementation detail)
- adds details about how `addNumLocales` works
- removes some redundancy in documentation, such as `:type:` annotations
- lots of wordsmithing / making things more concise

Closes https://github.com/chapel-lang/chapel/issues/14146

### CODE REVIEW

File: modules/packages/UnitTest.chpl [hunk 1]
```diff
@@ -19,19 +19,24 @@
  */
 
 /*
-Module UnitTest provides support for automated testing in Chapel.
-Any function of the form
+The UnitTest module provides support for automated testing in Chapel.
+
+The UnitTest module is intended to be used in with the `mason test
```
Comment (e-kayrakli) [inline modules/packages/UnitTest.chpl]: drop "in"?

File: modules/packages/UnitTest.chpl [hunk 2]
```diff
@@ -19,19 +19,24 @@
...
+<https://chapel-lang.org/docs/tools/mason/mason.html>`_ command, which
```
Comment (e-kayrakli) [inline modules/packages/UnitTest.chpl]: is the trailing _ a part of chpldoc syntax?

**Comment by ben-albrecht** [comment]
@e-kayrakli is reviewing, but wanted to ping @krishnadey30 as a heads up :)

File: modules/packages/UnitTest.chpl (hunk 2 above)
Comment (ben-albrecht) [reply modules/packages/UnitTest.chpl]: rst syntax specifically, but yes.

**Pull request merged**